quill-core/src/test/scala/io/getquill/dsl/MetaDslSpec.scala

```diff
@@ -70,10 +70,30 @@ class MetaDslSpec extends Spec {
           meta.extract(Row("a", Some("1"), None)) mustEqual (("a", None))
         }
       }
+      "optional deep nested" - {
+        case class Entity1(a: String, b: Int)
+        case class Entity2(a: Option[Int])
+
+        val meta = materializeQueryMeta[(String, Option[(Entity1, Entity2)])]
+        meta.expand.toString mustEqual "(q) => q.map(x => (x._1, x._2.map((v) => v._1.a), x._2.map((v) => v._1.b), x._2.map((v) => v._2.a)))"
+
+        "extracts Some if all columns are defined" in {
+          meta.extract(Row("a", Some("1"), Some(2), Some(Some(3)))) mustEqual
+            (("a", Some((Entity1("1", 2), Entity2(Some(3))))))
+        }
+        "extracts Some if optional column is undefined" in {
+          meta.extract(Row("a", Some("1"), Some(2), Some(Some(3)))) mustEqual
+            (("a", Some((Entity1("1", 2), Entity2(Some(3))))))
+        }
+        "extracts None if one column is undefined" in {
+          meta.extract(Row("a", Some("1"), None, Some(Some(3)))) mustEqual
```

Comment:
Not sure about `Some(Some(3))`. I think it should able to extract `Some(3)` but now it's failing. Will try to fix that.